src/App.tsx

```diff
@@ -1,39 +1,143 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
-import Carousel from './components/Carousel';
-
-interface State {
-  images: string[];
-}
-
-class App extends React.Component<{}, State> {
-  state = {
-    images: [
-      './img/1.png',
-      './img/2.png',
-      './img/3.png',
-      './img/4.png',
-      './img/5.png',
-      './img/6.png',
-      './img/7.png',
-      './img/8.png',
-      './img/9.png',
-      './img/10.png',
-    ],
+import { Carousel } from './components/Carousel';
+
+const images: string[] = [
+  './img/1.png',
+  './img/2.png',
+  './img/3.png',
+  './img/4.png',
+  './img/5.png',
+  './img/6.png',
+  './img/7.png',
+  './img/8.png',
+  './img/9.png',
+  './img/10.png',
+];
+
+const App: React.FC = () => {
+  const [itemWidth, setBoxWidth] = useState(130);
+  const [frameSize, setFrameSize] = useState(3);
+  const [step, setStep] = useState(3);
+  const [animationDuration, setAnimationDuration] = useState(1000);
+  const [infinite, setInfinite] = useState(false);
```

Comment:
make one state and one handler for all inputs. use event.target.name for change proper field in state.
const [inputs, setInputs] = useState({
```suggestion
const [inputs, setInputs] = useState({
  itemWidth:  130,
  frameSize: 3,
  ...
})
```
value for name attibute in input should be the same as field in inputs object.

Reply:
But why? We now have different processes running separately. Instead, we will get one object that changes completely for the one of inputs. Is not it?